build_and_push_images.py

```diff
@@ -77,30 +76,25 @@ def build_and_push_dockerfile(
             temp_dockerfile.name,
             "-t",
             target_tag,
+            "-t",
+            target_tag_major_min,
+            "--platform",
+            platform,
+            # We always push because docker buildx does not look at the local
+            # images, it always pull from repos.
+            "--push",
+            ".",
         ]
-        + (["-t", target_tag_major_min] if push else [])
-        + ["--platform", platform]
-        + (["--push"] if push else [])
-        + ["."]
     )
 
-    _run_command(build_command)
     temp_dockerfile.unlink()
 
 
-def main(platform: str, push: bool = False, beta: bool = False) -> None:
-    git_tag = os.getenv("GIT_TAG")
-    if not git_tag:
-        git_tag = (
-            subprocess.check_output(["git", "describe", "--abbrev=0", "--tags"])
-            .decode("utf-8")
-            .strip()
+def main(version: str, platform: str, beta: bool = False) -> None:
+    if not re.match(r"^v[^.]+\.[^.]+\.[^.]+$", version):
+        raise ValueError(
+            f"Expected version ({version}) be in vX.Y.Z format. It is not."
         )
-    print(f"Git tag: {git_tag}")
-
-    if not git_tag.startswith("v"):
-        print('Tag does not start with "v", ignoring')
-        sys.exit(0)
```

Comment:
You did change something here.
If the version doesn't start with a v you will raise an exception and the script will stop with a non 0 exit code.

Is that what you want ?

Reply:
I didn't notice the difference at first. I originally I thought that it was `sys.exit(1)`. I think that we're better off returning a non-zero exit code here so that our caller knows that something is wrong. This is the kind of error that we can't just ignore implicitly I think. If the caller doesn't care, they can ignore the exit code.